### PR TITLE
add opentip v2.4.6 with git auto-update

### PR DESCRIPTION
**Pull request by maruilian11**
**npm package url:** https://www.npmjs.com/package/opentip
**git repo url:** https://github.com/enyo/opentip
**Watch 94
  Star 1,165
 Fork 409**
@Amomo could u help me check this PR for #6187 ? thank you!
**i chose git auto-update rather than npm because git repo has 39 releases, but npm package only has 6 releases.**
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR 
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by Amomo** [comment]
@maruilian11 sorry for late.
I think this PR is ok, could you rebase it on our latest master branch?
Thank you!

**Comment by maruilian11** [comment]
@Amomo updated ~

**Comment by Amomo** [comment]
@PeterDaveHello I think this PR is ok.
[README.md](https://github.com/enyo/opentip/tree/v2.4.6#installation) of this repo said that users can take the files from `"lib/"` or `"downloads/"` based on they needs.
And, its GitHub repo contains the latest version and more versions than npm, so I think it is fine to use a git auto-update config here.

**Comment by PeterDaveHello** [comment]
Should point main file to `downloads/` ... and there is no need to minify `lib`/

**Comment by maruilian11** [comment]
@Amomo i have unminified the files under lib and point filename to a file under downloads,
could you help me check this PR again? thank you.

**Comment by Amomo** [comment]
@maruilian11 
I think we still need to minify `lib/opentip.js`, and use its minified file as main file.
Files in `downloads/` has combined with a framework, we may also host a minified version without binding any framework.

cc @PeterDaveHello , is this opinion ok?

**Comment by PeterDaveHello** [comment]
You should use `downloads`

**Comment by maruilian11** [comment]
@Amomo i don't really understand Peter's meaning, so i need to minify `/lib/opentip.js`, but point to `downloads`? or i don't need to do any change?

**Comment by PeterDaveHello** [comment]
There is no need to minify `lib`, read its document.

**Comment by maruilian11** [comment]
@Amomo could you help me check this PR again? thanks.

File: ajax/libs/opentip/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "bonzo": "1.x",
+    "reqwest": "0.x"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/enyo/opentip.git",
+    "basePath": "",
+    "files": [
+      "+(css|downloads|lib)/opentip*.+(js|css|map)",
+      "lib/adapter*.+(js|map)"
```
Comment (PeterDaveHello) [inline ajax/libs/opentip/package.json]: Don't we need `lib/opentip.js`?

Comment (maruilian11) [inline ajax/libs/opentip/package.json]: it's included by `+(css|downloads|lib)/opentip*.+(js|css|map)`.

Comment (maruilian11) [inline ajax/libs/opentip/package.json]: or i can change to `+(css|downloads|lib)/+(opentip|adapter)*.+(js|css|map).

**Comment by Amomo** [comment]
@maruilian11 
I think this PR is ok, could you rebase it on our latest master branch?
Thank you for update it several times!

**Comment by maruilian11** [comment]
@Amomo i rebased the branch with our latest master. could you help me check it again? thanks.

**Pull request merged**